#167142660 create icon atom

Sojisoyoye opened the pull request:
#### What does this PR do?

This PR creates icon field atom

#### Description of task to be completed?

- write test for the feature
- create style component for icon field
- create icon field component

#### How should this be manually tested?

- run npm install
- run npm run styleguide on your console

#### Any background context you want to provide?

-  install node

#### What are the relevant pivotal tracker stories?

[167142660](https://www.pivotaltracker.com/n/projects/2351184/stories/167142660)

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/43131939/61076527-5a222880-a414-11e9-96a2-dafae64a9091.png)

benfluleck commented on src/components/UI/atoms/Icon/Icon.jsx:
```diff
@@ -0,0 +1,84 @@
+  margin,
+  icon,
+}) => {
+  const IconType = icomoon[icons[icon]];
```
what is `icomoon`?

Sojisoyoye replied:
icomoon is the name of the icon pack from styled-icon component library.

benfluleck replied:
please rename since it is a default export for readability

Sojisoyoye replied:
ok

benfluleck commented on src/components/UI/atoms/Icon/Icon.jsx:
```diff
@@ -0,0 +1,84 @@
+  width,
+  display,
+  margin,
+  icon,
```
Please change to iconName I think that is better

benfluleck commented on src/components/UI/atoms/Icon/Icon.jsx:
```diff
@@ -0,0 +1,84 @@
+};
+
+Icon.defaultProps = {
+  id: 'mockIcon',
```
why would you use this as a defaultProp name ?

benfluleck commented on src/components/UI/atoms/Icon/Icon.jsx:
```diff
@@ -0,0 +1,84 @@
+
+Icon.defaultProps = {
+  id: 'mockIcon',
+  icon: 'search',
```
Why would you want your default iconName be search

Sojisoyoye replied:
The component requires that an iconName is supplied for it to render, as it is a required prop-type and search is one of the icons we need on the homepage so I decided to use it as default.

benfluleck commented on src/components/UI/atoms/Icon/Icon.spec.js:
```diff
@@ -0,0 +1,70 @@
+  it('should render the Icon with height', () => {
+    const { getByTestId } = setup({ height: 'medium' });
+
+    const icon = getByTestId('mockIcon');
```
Use queryByTestId https://testing-library.com/docs/dom-testing-library/api-queries

Sojisoyoye committed:
feat(icon atom): create icon atom

- create style component for icon field
- create icon field component
- implement test for feature

[Finishes #167142660]

Sojisoyoye committed:
feat(icon atom): create icon atom

- create style component for icon field
- create icon field component
- implement test for feature

[Finishes #167142660]

Sojisoyoye committed:
feat(icon atom): create icon atom

- create style component for icon field
- create icon field component
- implement test for feature

[Finishes #167142660]
